archive_viewer/table_models/curve_model.py

```diff
@@ -108,23 +136,27 @@ def set_data(self, column_name: str, curve: BasePlotCurveItem, value: Any) -> bo
                     return False
             elif value_is_formula and curve_is_formula:
                 try:
-                    rowName = self._row_names[index.row()]
-                    pv_dict = self.formulaToPVDict(rowName, value)
+                    pv_dict = self.formulaToPVDict(self._row_names[index.row()], value)
                     curve.formula = value
                     curve.pvs = pv_dict
                 except (SyntaxError, ValueError) as e:
                     logger.error(e)
                     return False
-            elif not value_is_formula and not curve_is_formula:
-                if value == curve.address:
-                    return True
-                logger.debug(f"Disconnecting old channel(s): {curve.address}")
-                [ch.disconnect() for ch in curve.channels() if ch]
-                curve.address = str(value)
-                logger.debug(f"Connecting new channel(s): {curve.address}")
-                [ch.connect() for ch in curve.channels() if ch]
-            else:
-                self.replaceToArchivePlot(curve=curve, index=index, address=value, color=curve.color)
+            elif not value_is_formula:
+                # Check if this thing already in curves model
+                if value in self:
+                    logger.warning("You can only have one of each PV")
+                    return False
+                if not curve_is_formula:
+                    if value == curve.address:
+                        return True
```

Comment:
Possible duplication of above check: `if value in self`
This makes the same check right? It can probably be deleted here

Reply:
Yes I guess it would be redundant but this is a innteresting thought - the first check is to make sure a new row doesnt have the same channel as another; the second is to save time if a row is set to the same channel as it already is. I suppose if somebody tries to set a row to itself, then it would just... do nothing?